CavalcanteLucas/todolist
# OL-39: Reset password

**Link to task:** 
https://labcodes.atlassian.net/browse/OL-39

**How to test:**
- With an user that was created, fill the form for password reset at `/password_reset`;
- Check if failure message appears when filling the “Email” field with invalid e-mail or not registered e-mail;
- If valid e-mail is provided, check if:
i. Redirection to `/` is occuring;
ii. Success message with instructions is appearing;
iii. E-mail has been sent with valid link to `/password/reset/confirm/:uid/:token`
- Check if failure message appears when the “New password” and “Confirm new password” fields are not valid or the same;
- In case of success, check if the hash for the users’ password has been altered.

**Description of your solution:**
With this PR one can change her/his login password. This PR enables React-based views and routes for password reset feature using the API provided by the library `django-rest-auth`. This request occurs in a 2-step cycle. In the first step, the user requires her/his password to changed. In the second step, the user enters a unique link to provide a new password. The user should not be able to submit neither the form to password reset nor the form to confirm password reset if those are not properly filled. Error handling is performed by the backend API. Successful requests should redirect to the index page and return a success message in both cases. The cycle is considered successful if the  user’s password has been changed.
